### ISSUE TITLE
Diff report generation in checkstyle-tester does not work properly

### ISSUE THREAD

**Issue by Vampire** [checkstyle/contribution#353]
When I start the checkstyle tester (with Java 8 as Java 11 did not work properly as mentioned in the other ticket), it goes through until it comes to the point where it wants to generate the diff report.
But when it tries to call the `java -jar $diffToolJarPath --patchReport $patchReport --output $outputDir --patchConfig $cfg.patchConfig --baseReport $baseReport --baseConfig $cfg.baseConfig` command this stacktrace is printed:
```
Starting diff report generation ...
Error: Registry key 'Software\JavaSoft\Java Runtime Environment'\CurrentVersion'
has value '1.8', but '1.7' is required.
Error: could not find java.dll
Error: Could not find Java SE Runtime Environment.
Caught: groovy.lang.GroovyRuntimeException: Error: !
groovy.lang.GroovyRuntimeException: Error: !
        at diff.executeCmd(diff.groovy:342)
        at diff.executeCmd(diff.groovy:336)
        at diff$_generateDiffReport_closure3.doCall(diff.groovy:234)
        at diff.generateDiffReport(diff.groovy:217)
        at diff$generateDiffReport$14.callCurrent(Unknown Source)
        at diff.run(diff.groovy:32)
```
Calling the same command (I used `println` to get the exact one for copy&paste) manually works fine. There has to be some flaw in the process creation of the tool or something like that.

**Comment by Vampire** [comment]
If I comment out the `getOsSpecificCmd` part it works for the report generation call, but then it fails on the `mvn` calls already.
For now I locally use an `executeRawCmd` for the report generation and the normal `executeCmd` for the other calls and now it runs through.

**Comment by rnveach** [comment]
I cannot reproduce this but googling looks like it might be to an invalid JAVA_HOME and multiple JVMs installed.
https://stackoverflow.com/questions/9313353/error-when-checking-java-version-could-not-find-java-dll

**Comment by Vampire** [comment]
Google finds many possible causes for this error from invalid JAVA_HOME to broken files and whatnot.
Yet I don't think any applies here.
In my path is `C:\ProgramData\Oracle\Java\javapath` where Java 9 executables are present.
Calling them like `java -version` from Cygwin Bash or cmd works fine.
Actually calling all `java.exe` files on my system works without a problem.
```
$ locate /java.exe | grep '/java\.exe$' | xargs -i sh -c '"{}" -version'
java version "1.8.0_51"
Java(TM) SE Runtime Environment (build 1.8.0_51-b16)
Java HotSpot(TM) 64-Bit Server VM (build 25.51-b03, mixed mode)
java version "1.7.0_80"
Java(TM) SE Runtime Environment (build 1.7.0_80-b15)
Java HotSpot(TM) 64-Bit Server VM (build 24.80-b11, mixed mode)
java version "1.7.0_80"
Java(TM) SE Runtime Environment (build 1.7.0_80-b15)
Java HotSpot(TM) 64-Bit Server VM (build 24.80-b11, mixed mode)
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)
java version "9.0.4"
Java(TM) SE Runtime Environment (build 9.0.4+11)
Java HotSpot(TM) 64-Bit Server VM (build 9.0.4+11, mixed mode)
java version "1.8.0_112"
Java(TM) SE Runtime Environment (build 1.8.0_112-b15)
Java HotSpot(TM) 64-Bit Server VM (build 25.112-b15, mixed mode)
java version "1.6.0_45"
Java(TM) SE Runtime Environment (build 1.6.0_45-b06)
Java HotSpot(TM) 64-Bit Server VM (build 20.45-b01, mixed mode)
java version "1.7.0_80"
Java(TM) SE Runtime Environment (build 1.7.0_80-b15)
Java HotSpot(TM) 64-Bit Server VM (build 24.80-b11, mixed mode)
java version "9.0.4"
Java(TM) SE Runtime Environment (build 9.0.4+11)
Java HotSpot(TM) 64-Bit Server VM (build 9.0.4+11, mixed mode)
openjdk version "1.8.0_152-release"
OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b39)
OpenJDK 64-Bit Server VM (build 25.152-b39, mixed mode)
openjdk version "1.8.0_152-release"
OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b28)
OpenJDK 64-Bit Server VM (build 25.152-b28, mixed mode)
java version "1.8.0_25"
Java(TM) SE Runtime Environment (build 1.8.0_25-b18)
Java HotSpot(TM) 64-Bit Server VM (build 25.25-b02, mixed mode)
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
java version "9.0.4"
Java(TM) SE Runtime Environment (build 9.0.4+11)
Java HotSpot(TM) 64-Bit Server VM (build 9.0.4+11, mixed mode)
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
openjdk version "10.0.2" 2018-07-17
OpenJDK Runtime Environment 18.3 (build 10.0.2+13)
OpenJDK 64-Bit Server VM 18.3 (build 10.0.2+13, mixed mode)
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
java version "1.8.0_40"
Java(TM) SE Runtime Environment (build 1.8.0_40-b25)
Java HotSpot(TM) 64-Bit Server VM (build 25.40-b25, mixed mode)
java version "1.8.0_25"
Java(TM) SE Runtime Environment (build 1.8.0_25-b18)
Java HotSpot(TM) 64-Bit Server VM (build 25.25-b02, mixed mode)
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
```
Yet if the `diff.groovy` calls it like it is doing, it fails.
I tried both, letting the `diff.groovy` call `java -version` and calling the command `diff.groovy` is calling from Cygwin Bash and cmd.exe.
The result was always the same.
It works properly manually, but how diff.groovy is doing it, it fails with this error.

**Comment by Vampire** [comment]
Ah, forget it I'm sorry.
There indeed were a Java 7 `java.exe` and `javaw.exe` in C:\Windows\System32 that were broken.
Strangely if I list this directory with Cygwin Bash, or even with cmd.exe started from Cygwin Bash, those files are just not there, I don't know why, damn strange Windows non-sense I guess.
When I started a cmd.exe not from Cygwin Bash it also failed with that error.
And now that I deleted those files, it works also from the diff.groovy.
I finally fired up Process Monitor to investigate the difference and found that actually different executable is used which just was not visible from Process Explorer, as the process died too fast to display anything.
